src/storage/kv/store.rs

```diff
@@ -25,41 +28,54 @@ use crate::storage::{
 };
 
 /// An MVCC-based transactional key-value store.
-#[derive(Clone)]
 pub struct Store {
     pub(crate) core: Arc<Core>,
+    pub(crate) is_closed: AtomicBool,
+    stop_tx: Sender<()>,
+    task_runner_handle: Arc<RwLock<Option<JoinHandle<()>>>>,
 }
 
 impl Store {
     /// Creates a new MVCC key-value store with the given options.
     /// It creates a new core with the options and wraps it in an atomic reference counter.
     /// It returns the store.
     pub fn new(opts: Options) -> Result<Self> {
-        let core = Arc::new(Core::new(opts)?);
-        Ok(Self { core })
+        // TODO: make this channel size configurable
+        let (writes_tx, writes_rx) = bounded(10000);
+        let (stop_tx, stop_rx) = bounded(1);
+
+        let core = Arc::new(Core::new(opts, writes_tx)?);
+        let task_runner_handle = TaskRunner::new(core.clone(), writes_rx, stop_rx).spawn();
+
+        Ok(Self {
+            core,
+            stop_tx,
+            is_closed: AtomicBool::new(false),
+            task_runner_handle: Arc::new(RwLock::new(Some(task_runner_handle))),
+        })
     }
 
     /// Begins a new read-write transaction.
     /// It creates a new transaction with the core and read-write mode, and sets the read timestamp from the oracle.
     /// It returns the transaction.
-    pub fn begin(&self) -> Result<Transaction> {
+    pub async fn begin(&self) -> Result<Transaction> {
```

Comment:
Can we perhaps keep this non-async, as none of the functions we call here are async?

Reply:
Totally agree, thank you for catching this

Reply:
There is one issue though, in the [mutex PR](https://github.com/surrealdb/surrealkv/pull/12) I am using async locks, because of which this this has to become async as taking a lock is also async. I am not really sure if I need the async locks though, but wdyt regarding that PR? If not required, would be easy to make this sync